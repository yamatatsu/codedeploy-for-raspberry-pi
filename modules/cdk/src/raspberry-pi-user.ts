import * as cdk from "@aws-cdk/core";
import * as iam from "@aws-cdk/aws-iam";

export class RaspberryPiUser extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    new iam.User(this, "User", {
      managedPolicies: [
        iam.ManagedPolicy.fromAwsManagedPolicyName("AWSCodeDeployFullAccess"),
      ],
    });
  }
}
