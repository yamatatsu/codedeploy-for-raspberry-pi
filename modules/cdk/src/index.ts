import "source-map-support/register";
import * as cdk from "@aws-cdk/core";
import { RaspberryPiUser } from "./raspberry-pi-user";

const app = new cdk.App();
new RaspberryPiUser(app, "TvKesu-RaspberryPiUser", {});
