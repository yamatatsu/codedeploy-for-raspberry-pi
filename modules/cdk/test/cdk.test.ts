import { SynthUtils } from "@aws-cdk/assert";
import * as cdk from "@aws-cdk/core";
import { RaspberryPiUser } from "../src/raspberry-pi-user";

test("RaspberryPiUser", () => {
  const app = new cdk.App();

  const target = new RaspberryPiUser(app, "Target");

  expect(SynthUtils.toCloudFormation(target)).toMatchSnapshot();
});
