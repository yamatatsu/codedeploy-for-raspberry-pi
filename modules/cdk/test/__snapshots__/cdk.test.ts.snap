// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RaspberryPiUser 1`] = `
Object {
  "Resources": Object {
    "User00B015A1": Object {
      "Properties": Object {
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/AWSCodeDeployFullAccess",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::User",
    },
  },
}
`;
